frontend/pages/g/_groupSlug/r/create/bookmarklet.vue

```diff
@@ -0,0 +1,118 @@
+            :prepend-inner-icon="$globals.icons.tags"
+            filled
+            rows="2"
+            class="rounded-lg mt-2"
+            rounded
+            :hint="$t('recipe.create-bookmarklet-hint')"
+            persistent-hint
+            :readonly="true"
+            @click="copyBookmarklet"
+          ></v-textarea>
+        </div>
+      </v-form>
+    </div>
+  </template>
+
+  <script lang="ts">
+  import {
+    defineComponent,
+    reactive,
+    toRefs,
+    ref,
+    useRouter,
+    computed,
+    useContext,
+    useRoute
+  } from "@nuxtjs/composition-api";
+  import { detectServerBaseUrl } from "~/composables/use-utils";
+  import { VForm } from "~/types/vuetify";
+  import { useCopy } from "~/composables/use-copy";
+
+  export default defineComponent({
+    setup() {
+      const { $auth, req, i18n } = useContext();
+
+      const { copyText } = useCopy();
+
+      const route = useRoute();
+
+      const router = useRouter();
+
+      const importKeywordsAsTags = computed({
+        get() {
+          return route.value.query.use_keywords === "1";
+        },
+        set(v: boolean) {
+          router.replace({ query: { ...route.value.query, use_keywords: v ? "1" : "0" } });
+        },
+      });
+
+      const stayInEditMode = computed({
+        get() {
+          return route.value.query.edit === "1";
+        },
+        set(v: boolean) {
+          router.replace({ query: { ...route.value.query, edit: v ? "1" : "0" } });
+        },
+      });
+
+      const baseUrl = computed(() => detectServerBaseUrl(req));
+      const groupSlug = computed(() => route.value.params.groupSlug || $auth.user?.groupSlug || "");
+      const importKeywordsAsTagAsNumber = computed(() => Number(importKeywordsAsTags.value));
+      const stayInEditModeAsNumber = computed(() => Number(stayInEditMode.value));
+
+      const bookmarkletResult = computed({
+        get() {
+        const route = useRoute();
+
+        let url = document.URL;
+        let slashCount = 0;
+        let position = -1;
+
+        // The third slash is after the port portion of the URL
+        while(slashCount < 3) {
+          position = url.indexOf("/", position + 1);
+          if (position === -1) {
+            break;
+          }
+          slashCount++;
+        }
+
+        // Now that we found the third slash, remove it and everything after
+        url = url.substring(0, position);
+
+        // window.history.replaceState is appended to fix Vivaldi bug
+        // https://forum.vivaldi.net/topic/31409/bookmarklets-replaces-the-url-in-the-address-bar/25?lang=en-US&page=2
+        return "javascript:" + encodeURIComponent(`(function(){var dest="${url}/g/${groupSlug.value}/r/create/url?use_keywords=${importKeywordsAsTagAsNumber.value}&edit=${stayInEditModeAsNumber.value}&recipe_import_url="+encodeURIComponent(document.URL);window.open(dest,"_blank");window.history.replaceState({},"",location.href);})()`) + ";"
+      }
+      });
```

Comment:
the easier way to do this would be to use the URL origin of the page
Furthermore you can just cast the booleans (eg. importKeywordsAsTag) as Number without having to add a additional variables
```suggestion
      const bookmarkletResult: ComputedRef<string> = computed(() => {
        // get the origin of the current page
        const url = new URL(document.URL).origin;

        // window.history.replaceState is appended to fix Vivaldi bug
        // https://forum.vivaldi.net/topic/31409/bookmarklets-replaces-the-url-in-the-address-bar/25?lang=en-US&page=2
        return "javascript:" + encodeURIComponent(`(function(){var dest="${url}/g/${groupSlug.value}/r/create/url?use_keywords=${Number(importKeywordsAsTags.value)}&edit=${Number(stayInEditMode.value)}&recipe_import_url="+encodeURIComponent(document.URL);window.open(dest,"_blank");window.history.replaceState({},"",location.href);})()`) + ";"
      });
```